### ISSUE TITLE
Beschrijving bij /bewoners niet juist en duidelijk voor periode

### ISSUE THREAD

**Issue by fsamwel** [VNG-Realisatie/Haal-Centraal-BRP-bevragen#312]
Voor operatie /historie/bewoningen staat in de description:

> Het resultaat bevat de verblijfsplaatshistorie waarvan de EINDdatumgeldigheidverblijfsplaats op of na de datumVan ligt EN de BEGINdatum op of voor de DatumTotEnMet ligt.
> 
> Of
> 
> Het resultaat bevat de verblijfsplaats historie waarvan de begindatumgeldigheidverblijfsplaats voor op op de geldigOp ligt en de eindatumgeldigheidverblijfsplaats na of op de geldigOp ligt.
> 
> Als geldigOp is ingevuld mogen datumVan en datumTotenMet nietingevuld zijn.

Dit is niet in lijn met de parameters die zijn gedefinieerd: geldigvan, geldigtot, geldigop.

**Comment by fsamwel** [comment]
Ook de beschrijvingen bij de parameters geldigvan, geldigtot, geldigop bevatten fouten en zouden duidelijker kunnen:
geldigop: "De peildatum **van de periode** waarover de bewoning wordt opgevraagd. Het resultaat bevat de bewoning en de bewoners waarvan de **begindatumgeldigheidDe peildatum** van de periode waarover de **verblijfsplaatshistorie** wordt opgevraagd."

geldigtot: "... **verblijfsplaatshistorie** waarvan de **einddatum** op of na de **datumVan** ligt"

De response bevat bovendien de geldigheid als geldigTotEnmet, terwijl de vraagparameter geldigtot is. Is het niet duidelijker consequent tot te gebruiken of consequent tot en met?

geldigvan: "De begindatum van de periode waarover de bewoning wordt opgevraagd. Het resultaat bevat **debewoningen** en bewoners waarvan de begindatum op of na de **datumVan** ligt"

Ik denk dat het duidelijker is wanneer consequent alleen namen van properties in de response (dan wel namen van elementen in LO GBA) en namen van parameters in de rekest worden gebruikt.

**Comment by JohanBoer** [comment]
Inmiddels zijn de namen en beschrijvingen van de parameters generiek gemaakt en in Common.yaml geplaatst. Beschrijving is ook aangepast.

**Comment by fsamwel** [comment]
@JohanBoer ik ben het niet eens met de beschrijving "Als peildatum is ingevuld mogen datumVan en datumtotenmet nietingevuld zijn."
Zo hebben we dat ook niet gedaan in de historie endpoints. Het zijn gewoon queryparameters dus:
- als peildatum binnen de periode ligt wordt de situatie op peildatum teruggegeven
- als peildatum buiten de periode ligt worden geen bewoners teruggegeven

Kan je deze zin verwijderen uit de description

**Comment by fsamwel** [comment]
idem voor "Als zowel peildatum, datumVan en datumtotenmet niet ingevuld zijn wordt impliciet de datum van vandaag gebruikt als peildatum."
Ook dit is niet consistent met de historie endpoints. Als je query parameters niet invult, wordt er niet op gefilterd. Dus de standaard RESTful werking is dat bij weglaten van zowel periode als peildatum de gehele bewoninghistorie wordt teruggegeven.
Ik zie eigenlijk geen reden van de standaard RESTful werkwijze af te wijken.

@CathyDingemanse is deze specifieke afhandeling, defaultpeildatum is vandaag, functioneel gewenst?

**Comment by CathyDingemanse** [comment]
Nee, volgens mij niet. De actuele bewoning kan al met de ingeschreven persoon worden verkregen. 

Dat gezegd hebbende: het is wel zo dat zonder opgave van periode of peildatum altijd de hele doopzeel wordt geleverd van bewoning op hetzelfde adres als BSN (dat zijn dus alle bewoner van alle adressen waar iemand ooit heeft gewoond sinds 1993, of de hele historische bewoning van het adres (tot 1993). 

Ik denk niet dat iedere gebruiker door heeft wat het opgeven van een simpele identificatie nummeraanduiding of BSN voor consequenties kan hebben. 

Zullen we het opgeven van een periode of peildatum verplicht maken? Dat is ook goed voor de bewustwording van consumer (wat vraag ik eigenlijk?).

**Comment by JohanBoer** [comment]
De beschrijving wordt : 

Op basis van de parameters wordt een of meerdere bewoningen geretourneerd inclusief gegevens over de bewoners van het betreffende verblijfobject. Ten minste één van de volgende parameters moet zijn opgenomen.

Identificatiecodenummeraanduiding
Burgerservicenummer

De bovenstaande parameters moeten gecombineerd worden met een periode of een peildatum.

Overleden personen worden opgenomen in het zoekresultaat tot aan de datum van overlijden voor zover deze datum overlijden binnen de opgevraagde periode valt. .

Het resultaat bevat de adressen en bewoners van een verblijfplaats waarvan de EINDdatum geldigheidverblijfplaats op of na de datumvan ligt EN de BEGINdatum op of voor de datumtotenmet ligt.

Of

Het resultaat bevat de het adres en de bewoners van de verblijfplaats waarbij de BEGINdatumgeldigheidverblijfplaats voor op op de peildatum ligt en de EINDdatumgeldigheidverblijfplaats na of op de peildatum ligt.

Er wordt niet op de gevonden adressen gesorteerd, maar per adres (bewoning) worden de bewoners wel gesorteerd op datumvan.

**Comment by fsamwel** [comment]
> Het resultaat bevat de adressen en bewoners van een verblijfplaats waarvan de EINDdatum geldigheidverblijfplaats op of na de datumvan ligt EN de BEGINdatum op of voor de datumtotenmet ligt.
Of
Het resultaat bevat de het adres en de bewoners van de verblijfplaats waarbij de BEGINdatumgeldigheidverblijfplaats voor op op de peildatum ligt en de EINDdatumgeldigheidverblijfplaats na of op de peildatum ligt.

Ik zou de "Of" hier tussenuit willen, want het is geen of. (eigenlijk is en een En)

> De gevonden adressen worden aflopend gesorteerd, en per adres (bewoning) worden de bewoners wel aflopend gesorteerd op datumvan.
Willen we dit zo? Als ik bewoningen vraag op burgerservicenummer, is het toch logisch dat het meest actuele adres van deze bewoner bovenaan komt?
Bij het sorteren op bewoners, zou ik ook sorteren op datumTot

**Comment by JohanBoer** [comment]
Het moet inderdaad "en" zijn. Als de peildatum buiten de opgegeven periode valt hoor je geen resultaat krijgen. 

Deze tekst moet niet in de description staan, maar (het is provider-logica) in een feature.
